### PR TITLE
253 254 synchro and spdz element

### PR DESCRIPTION
**Pull request by n1v0lg**
First part of SPDZ revamp; removes unnecessary SpdzElement class (#253) and updates round synchronization to be more consistent with Spdz2k (#254).

### CODE REVIEW

File: suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/TestSpdzMisc.java [hunk 1]
```diff
@@ -25,18 +22,19 @@ public void testPreproStrat() {
...
   }
 
-  @Test(expected = RuntimeException.class)
-  public void testResourcePoolStoreNotInitialized() {
-    SpdzStorage store = new SpdzStorageImpl(
-        new SpdzStorageDataSupplier(new FilebasedStreamedStorageImpl(new InMemoryStorage()), "null",
-            2));
-    new SpdzResourcePoolImpl(1, 2, store);
-  }
+//  @Test(expected = RuntimeException.class)
```
Comment (Rhaea) [inline suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/TestSpdzMisc.java]: Remove or reintroduce if still relevant

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java [hunk 2]
```diff
@@ -2,23 +2,130 @@
 
 import dk.alexandra.fresco.framework.value.SInt;
 import java.io.Serializable;
+import java.math.BigInteger;
 
 public class SpdzSInt implements SInt, Serializable {
 
-  private static final long serialVersionUID = -9048612329603301195L;
-  public SpdzElement value;
+  private static final long serialVersionUID = 8828769687281856043L;
```
Comment (GuutBoy) [inline suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: Do we actually use this as a serializable? we may not need to implement that interface.

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java [hunk 3]
```diff
@@ -2,23 +2,130 @@
...
+  /**
+   * Create a SpdzSInt containing a share, mac and modulus. This constructor handles serialized
+   * data.
+   *
+   * @param data Array of bytes containing the share and mac. First half contains the share, second
+   * contains the mac
+   * @param modulus The modulus
+   * @param modulusSize The size of the share and mac
+   */
+  public SpdzSInt(byte[] data, BigInteger modulus, int modulusSize) {
```
Comment (GuutBoy) [inline suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: We should be able to compute the modulus size from the modulus it self. Getting it as a parameter seems to be just asking for problems.

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java [hunk 4]
```diff
@@ -2,23 +2,130 @@
...
+    BigInteger share = this.share.add(e.getShare()).mod(mod);
+    BigInteger mac = this.mac.add(e.getMac()).mod(mod);
+    return new SpdzSInt(share, mac, this.mod);
+  }
+
+  /**
+   * Add a public value to this element.
+   *
+   * @param e The element to add
+   * @param id The id
```
Comment (GuutBoy) [inline suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: What id?

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java [hunk 5]
```diff
@@ -2,23 +2,130 @@
...
+   * @return The sum
+   */
+  public SpdzSInt add(SpdzSInt e) {
+    BigInteger share = this.share.add(e.getShare()).mod(mod);
+    BigInteger mac = this.mac.add(e.getMac()).mod(mod);
+    return new SpdzSInt(share, mac, this.mod);
+  }
+
+  /**
+   * Add a public value to this element.
```
Comment (GuutBoy) [inline suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: element?

File: suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/maccheck/MaliciousSpdzMacCheckProtocol.java [hunk 6]
```diff
@@ -21,111 +22,74 @@
...
-
-        int t = as.size();
-
-        BigInteger[] rs = new BigInteger[t];
-        BigInteger temporaryR = s;
-        for (int i = 0; i < t; i++) {
-          temporaryR = new BigInteger(digest.digest(temporaryR.toByteArray())).mod(modulus);
-          rs[i] = temporaryR;
-        }
+        System.out.println("Malicious size " + openedValues.size());
```
Comment (GuutBoy) [inline suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/maccheck/MaliciousSpdzMacCheckProtocol.java]: please remove printout

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java (hunk 5 above)
Comment (GuutBoy) [reply suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: This seems to be some problems with copy pasting the java doc from the old SpdzElement class.

File: suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java [hunk 7]
```diff
@@ -16,38 +16,13 @@
    *
    * @param share The share
    * @param mac The mac
-   * @param modulus The modulus
```
Comment (GuutBoy) [inline suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java]: This should not be removed.

File: core/src/main/java/dk/alexandra/fresco/framework/util/OpenedValueStore.java [hunk 8]
```diff
@@ -22,15 +21,12 @@
   void pushOpenedValues(List<AuthT> newSharesWithMacs, List<OpenT> newOpenedValues);
 
   /**
-   * Default call to {@link #pushOpenedValues(List, List)} that wraps single elements in lists.
+   * Call to {@link #pushOpenedValues(List, List)} that wraps single elements in lists.
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/framework/util/OpenedValueStore.java]: This comment is no longer accurate.

**Pull request merged**